### PR TITLE
[GEOS-10368] Logging controller is added

### PR DESCRIPTION
**Pull request by ydeliorman**
[GEOS-10368](https://osgeo-org.atlassian.net/browse/GEOS-10368)

Logging controller is added to the /logging endpoint. It fetches and updates logging info using below json syntax.

```
{
    "logging": {
        "level": "GEOSERVER_DEVELOPER_LOGGING.properties",
        "location": "logs/geoserver2.log",
        "stdOutLogging": true
    }
}
```
  

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [x] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [x] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

File: doc/en/api/1.0.0/logging.yaml [hunk 1]
```diff
@@ -0,0 +1,94 @@
+---
```
Comment (aaime) [inline doc/en/api/1.0.0/logging.yaml:1]: Good, but many users will land on this one: https://docs.geoserver.org/latest/en/user/rest/api/index.html
Could you update it as well?

**Review by aaime — CHANGES_REQUESTED** [review]

File: src/restconfig/src/test/java/org/geoserver/rest/LoggingControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        global.getCoverageAccess().setQueueType(CoverageAccessInfo.QueueType.UNBOUNDED);
+        getGeoServer().save(global);
+
+        revertSettings(null);
+
+        // revert local settings
+        revertSettings("sf");
+    }
+
+    public void initLogging() {
```
[the remark this answers is not in the document]
Comment (ydeliorman) [reply src/restconfig/src/test/java/org/geoserver/rest/LoggingControllerTest.java]: done

[the remark this answers is not in the document]
Comment (ydeliorman) [reply src/restconfig/src/test/java/org/geoserver/rest/LoggingControllerTest.java]: done

File: doc/en/api/1.0.0/logging.yaml (hunk 1 above)
Comment (ydeliorman) [reply doc/en/api/1.0.0/logging.yaml:1]: done

File: src/restconfig/src/main/java/org/geoserver/rest/LoggingController.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
+    @Override
+    protected <T> ObjectWrapper createObjectWrapper(Class<T> clazz) {
+        return new ObjectToMapWrapper<>(
+                clazz, Arrays.asList(JAIInfo.class, CoverageAccessInfo.class));
+    }
+
+    @Override
+    public void configurePersister(XStreamPersister persister, XStreamMessageConverter converter) {
```
[the remark this answers is not in the document]
Comment (ydeliorman) [reply src/restconfig/src/main/java/org/geoserver/rest/LoggingController.java]: done

File: src/restconfig/src/test/java/org/geoserver/rest/LoggingControllerTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (ydeliorman) [reply src/restconfig/src/test/java/org/geoserver/rest/LoggingControllerTest.java]: Removed unnecessary parts but creating a GeoServer instance is still there due to having a default loggingInfo for get request tests.

File: src/restconfig/src/test/java/org/geoserver/rest/LoggingControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+public class LoggingControllerTest extends CatalogRESTTestSupport {
+
+    protected GeoServer geoServer;
+
+    @Before
+    public void init() {
+        geoServer = getGeoServer();
+
+        // revert global settings
+        GeoServerInfo global = getGeoServer().getGlobal();
```
[the remark this answers is not in the document]
Comment (ydeliorman) [reply src/restconfig/src/test/java/org/geoserver/rest/LoggingControllerTest.java]: Removed unnecessary parts but creating a GeoServer instance is still there due to having a default loggingInfo for get request tests.

**Review by aaime — APPROVED** [review]

**Comment by aaime** [comment]
Thanks, merged.

**Comment by jodygarnett** [comment]
Thanks for this contribution, since is a new feature would you like to thank your employer or customer?

For new functionality we try and tanks the organizations responsible in our release anouncements (and presentations).

**Comment by jodygarnett** [comment]
Testing 2.21-RC and I noticed that this controler does not require authentication, and may retrieve log location (which can contain an absolute path).

Can this end-point be secured please?

**Pull request merged**